src/config.rs

```diff
@@ -0,0 +1,266 @@
+use super::internal::{Bounded, Infinite, SizeLimit};
+use ::error::Result;
+use byteorder::{BigEndian, ByteOrder, LittleEndian, NativeEndian};
+use serde;
+use std::io::{Write, Read};
+use std::marker::PhantomData;
+
+use self::LimitOption::*;
+use self::EndianOption::*;
+
+struct DefaultOptions(Infinite);
+
+pub(crate) trait Options {
+    type Limit: SizeLimit + 'static;
+    type Endian: ByteOrder + 'static;
+
+    #[inline(always)]
```

Comment:
I think the inline attribute only means anything on functions that have a body. I don't think this automatically applies to all implementations of the trait (but I could be wrong).

Reply:
Done.

Reply:
No effect.

Reply:
Let's remove it then. It is still present in the most recent revision.

Reply:
Oh yeah, I accidentally removed them from the default methods on OptionsExt instead.  I wonder if they are used on default methods?

I'll look into this a bit more.